apps/experiments/views/experiment.py

```diff
@@ -161,19 +167,46 @@ def get_queryset(self):
             .filter(team=self.request.team, experiment__id=self.kwargs["experiment_id"])
             .select_related("participant__user")
         )
-
         if not self.request.GET.get("show-all"):
             query_set = query_set.exclude(experiment_channel__platform=ChannelPlatform.API)
+        query_set = self.apply_dynamic_filters(query_set)
+        return query_set
+
+    def apply_dynamic_filters(self, query_set):
+        filter_conditions = Q()
+        filter_applied = False
 
-        if tags_query := self.request.GET.get("tags"):
-            tags = tags_query.split("&")
-            query_set = query_set.filter(chat__tags__name__in=tags).distinct()
+        for i in range(30):  # arbitrary number higher than any # of filters we'd expect
```

Comment:
we could also do a while loop here and increment the index variable. Something like
```python
idx = 0
while True:
   ...
   idx += 1
```

Reply:
It was originally a while loop. I suggested changing it to a for loop to avoid any chance of runaway

Reply:
what about doing
```python
while filter_column and filter_operator and filter_value:
   ...
```